consensus/spos/bls/v2/subroundBlock.go

```diff
@@ -519,12 +519,23 @@ func (sr *subroundBlock) receivedBlockHeader(headerHandler data.HeaderHandler) {
 	ctx, cancel := context.WithTimeout(context.Background(), sr.RoundHandler().TimeDuration())
 	defer cancel()
 
-	_ = sr.processReceivedBlock(ctx, int64(headerHandler.GetRound()), []byte(sr.Leader()))
+	blockProcessedWithSuccess := sr.processReceivedBlock(ctx, int64(headerHandler.GetRound()), []byte(sr.Leader()))
 	sr.PeerHonestyHandler().ChangeScore(
 		sr.Leader(),
 		spos.GetConsensusTopicID(sr.ShardCoordinator()),
 		spos.LeaderPeerHonestyIncreaseFactor,
 	)
+
+	pubKeys := sr.ConsensusGroup()
+	if blockProcessedWithSuccess {
+		// increment the number of rounds without received message
+		// at this point, it is certain that the leader is not managed by self and
+		// the block was processed, so signatures are expected from the current managed nodes
+		for _, pk := range pubKeys {
+			pkBytes := []byte(pk)
+			sr.IncrementRoundsWithoutReceivedMessages(pkBytes)
```

Comment:
I think this is not working as expected because from the perspective of a backup machine, it does not exit at line 469, and then it could increment the rounds even if it has already increased on subround start round, so ending up with a double increase.